py/client/pydeephaven/table.py

```diff
@@ -804,3 +804,79 @@ def delete(self, table: Table) -> None:
             self.session.input_table_service.delete(self, table)
         except Exception as e:
             raise DHError("delete data in the InputTable failed.") from e
+
+
+class MultiJoinTable:
+    """A MultiJoinTable is an object that contains the result of a multi-table natural join. To retrieve the underlying
+    result Table, use the table() method. """
+
+    def __init__(self, table: Table):
+        self._table = table
+
+    def table(self) -> Table:
+        """Returns the Table containing the multi-table natural join output. """
+        return self._table
+
+
+class MultiJoinInput:
+    """A MultiJoinInput represents the input tables, key columns and additional columns to be used in the multi-table
+    natural join.
+    """
+    table: Table
+    on: Union[str, Sequence[str]]
+    joins: Union[str, Sequence[str]] = None
+
+    def __init__(self, table: Table, on: Union[str, Sequence[str]], joins: Union[str, Sequence[str]] = None):
+        """Initializes a MultiJoinInput object.
+
+        Args:
+            table (Table): the right table to include in the join
+            on (Union[str, Sequence[str]]): the column(s) to match, can be a common name or an equality expression that
+                matches every input table, i.e. "col_a = col_b" to rename output column names.
+            joins (Union[str, Sequence[str]], optional): the column(s) to be added from the table to the result
+                table, can be renaming expressions, i.e. "new_col = col"; default is None
+        """
+        self.table = table
+        self.on = to_list(on)
+        self.joins = to_list(joins)
+
+
+def multi_join(input: Union[Table, Sequence[Table], MultiJoinInput, Sequence[MultiJoinInput]],
+               on: Union[str, Sequence[str]] = None) -> MultiJoinTable:
+    """ The multi_join method creates a new table by performing a multi-table natural join on the input tables. The
+    result consists of the set of distinct keys from the input tables natural joined to each input table. Input
+    tables need not have a matching row for each key, but they may not have multiple matching rows for a given key.
+
+    Args:
+        input (Union[Table, Sequence[Table], MultiJoinInput, Sequence[MultiJoinInput]]): the input objects specifying the
+            tables and columns to include in the join.
+        on (Union[str, Sequence[str]], optional): the column(s) to match, can be a common name or an equality expression
+            that matches every input table, i.e. "col_a = col_b" to rename output column names. Note: When
+            MultiJoinInput objects are supplied, this parameter must be omitted.
+
+    Returns:
+        MultiJoinTable: the result of the multi-table natural join operation.
```

Comment:
```suggestion
        MultiJoinTable: the result of the multi-table natural join operation.  To access the underlying Table, use the
            table() method.
```
This is from the server.

Reply:
```suggestion
        MultiJoinTable: the result of the multi-table natural join operation.  To access the underlying Table, use the
            :meth:`~MultiJoinTable.table` method.
```

If this is the right way to link, it is needed in the server doc line as well.